src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -0,0 +1,124 @@
+
+    public event EventHandler<RequestShutdownEventArgs>? RequestServerShutdown;
+
+    public RequestExecutionQueue(
+        string serverKind,
+        ILspLogger logger)
+    {
+        throw new NotImplementedException();
+    }
+
+    public void Shutdown()
+    {
+        throw new NotImplementedException();
+    }
+
+    public void Start(ILspServices lspServices)
+    {
+        throw new NotImplementedException();
+    }
+
+
+    public Task ExecuteAsync(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        INotificationHandler<RequestContextType> handler,
+        string methodName,
+        ILspServices lspServices,
+        CancellationToken requestCancellationToken)
+    {
+        throw new NotImplementedException();
+    }
+
+    public Task ExecuteAsync<TRequestType>(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        INotificationHandler<TRequestType, RequestContextType> handler,
+        TRequestType request,
+        string methodName,
+        ILspServices lspServices,
+        CancellationToken requestCancellationToken)
+    {
+        throw new NotImplementedException();
+    }
+    
+    public Task<TResponseType> ExecuteAsync<TRequestType, TResponseType>(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        IRequestHandler<TRequestType, TResponseType, RequestContextType> handler,
+        TRequestType request,
+        string methodName,
+        ILspServices lspServices,
+        CancellationToken requestCancellationToken)
+    {
+        throw new NotImplementedException();
+    }
+
+    public virtual (IQueueItem<RequestContextType>, Task<TResponseType>) CreateQueueItem<TRequestType, TResponseType>(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        string methodName,
+        object? textDocument,
+        TRequestType request,
+        IRequestHandler<TRequestType, TResponseType, RequestContextType> handler,
+        ILspServices lspServices,
+        CancellationToken cancellationToken)
+    {
+        throw new NotImplementedException();
+    }
+
+    public virtual (IQueueItem<RequestContextType>, Task) CreateQueueItem<TRequestType>(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        string methodName,
+        object? textDocument,
+        TRequestType request,
+        INotificationHandler<TRequestType, RequestContextType> handler,
+        ILspServices lspServices,
+        CancellationToken cancellationToken)
+    {
+        throw new NotImplementedException();
+    }
+
+    public virtual (IQueueItem<RequestContextType>, Task) CreateQueueItem(
+        bool mutatesSolutionState,
+        bool requiresLSPSolution,
+        string methodName,
+        object? textDocument,
+        INotificationHandler<RequestContextType> handler,
+        ILspServices lspServices,
+        CancellationToken cancellationToken)
```

Comment:
a lot of thes emethods seems to pass along a whole lot of common stuff over and over again.  DOes this indicate a common sort of type that should be extracted for thsi concept?